### ISSUE TITLE
[Java] Implement new Concept Exercise: chars

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1964]
This issue describes how to implement the `chars` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Chars in Java.

**_NOTE:_** This may be forked from the [C# chars](https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/chars) exercise.

## Learning objectives

- Know of the existence of the `char` type.
- Know what a char represents (a Unicode character).
- Know how to define a char.
- Know how to access a char in a string by index.
- Know of some basic char methods (like converting to uppercase).
- Know that chars are immutable.
- Know how to use a `for` loop.

## Out of scope

- Converting an integer to a character and vice versa.

## Concepts

This Concepts Exercise's Concepts are:

- `chars`: know of the existence of the `char` type; know that a char represents; know how to define a char; know how to access a char in a string by index; know of some basic char methods (like converting to uppercase).
- `for-loops`: know how to use a `for` loop.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `strings`: know of the `string` type that will be iterated over and accessed by index.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer).

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer).

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by ystromm** [comment]
Hiya!
I made an attempt at porting the C# chars exercise. Should I create a PR?

**Comment by mirkoperillo** [comment]
Hi @ystromm thank you for helping the Java track.

Feel free to provide a PR, we will try to start the review ASAP

**Comment by mirkoperillo** [comment]
Fixed by PR #2144
